platform/src/main/java/org/stellar/anchor/platform/custody/Sep31CustodyPaymentHandler.java

```diff
@@ -0,0 +1,62 @@
+package org.stellar.anchor.platform.custody;
+
+import static org.stellar.anchor.util.Log.infoF;
+import static org.stellar.anchor.util.Log.warn;
+
+import io.micrometer.core.instrument.Metrics;
+import java.io.IOException;
+import org.stellar.anchor.api.exception.AnchorException;
+import org.stellar.anchor.api.sep.SepTransactionStatus;
+import org.stellar.anchor.apiclient.PlatformApiClient;
+import org.stellar.anchor.platform.custody.CustodyPayment.CustodyPaymentStatus;
+import org.stellar.anchor.platform.data.JdbcCustodyTransaction;
+import org.stellar.anchor.platform.data.JdbcCustodyTransactionRepo;
+import org.stellar.anchor.platform.service.AnchorMetrics;
+
+public class Sep31CustodyPaymentHandler extends CustodyPaymentHandler {
+
+  public Sep31CustodyPaymentHandler(
+      JdbcCustodyTransactionRepo custodyTransactionRepo, PlatformApiClient platformApiClient) {
+    super(custodyTransactionRepo, platformApiClient);
+  }
+
+  @Override
+  public void onReceived(JdbcCustodyTransaction txn, CustodyPayment payment)
+      throws AnchorException, IOException {
+    infoF(
+        "Incoming inbound payment for SEP-31 transaction. Payment: id[{}], externalTxId[{}]",
+        payment.getId(),
+        payment.getExternalTxId());
+
+    if (CustodyPaymentStatus.SUCCESS == payment.getStatus()) {
```

Comment:
Can we move `if (CustodyPaymentStatus.SUCCESS == payment.getStatus()) {` into `validatePayment` method? Just to not duplicate this validation 3 times